### DISCUSSION TITLE
Blazor App share state with CSLA Edit object

### DISCUSSION THREAD

**Discussion by adrianwright109** [MarimerLLC/csla#2169] [Questions]
I have created a Blazor Web Assembly application which is hosted in an ASP.NET Core Server - this is based on the CSLA [BlazorExample](https://github.com/MarimerLLC/csla/tree/main/Samples/BlazorExample) project.

I want to be able to share state (gloval values) between the Blazor Web Assembly `Client` and the CSLA objects in the `Shared` project.

For example I have a `ClientEdit` object in the `Shared` project which has a `BusinessRule` called `HasFeatureOption` that checks to see that the authenticated user has access to the particular application area (feature option).  

For the check to take place it needs the `[ClientId]` and `[ClientEnvionment]` for the authenticated user in their current browser session.

These values are not apart of the `ClientEdit` model and are additional info that I need to have access to.

```
        protected override void AddBusinessRules()
        {
            base.AddBusinessRules();

            BusinessRules.AddDataAnnotations();

            //permission checks
            BusinessRules.AddRule(new IsInUserRole(UserRole.AdminUser));

            //feature option checks - make sure user can access this area of the application
            BusinessRules.AddRule(new HasFeatureOption([ClientId], [ClientEnvironment], FeatureOption.Area1));
        }
```

Is it possible to share user application state between the Blazor `Client` and the CSLA objects - if so how best to achieve this?

**Comment by rockfordlhotka** [comment]
The `Csla.ApplicationContext.ClientContext` dictionary flows from client to server on each data portal call, and so this is the normal way to share state as you describe.

We used to also have `GlobalContext` which was bi-directional, but that's now marked obsolete and is going away, because in a multi-threaded parallel world the merging of multiple concurrent server-side changes back into the client is ... challenging.

Be aware that there's a [known bug with Blazor wasm and `ClientContext`](https://github.com/MarimerLLC/csla/issues/2161) right now. The fix is in the next release.
